definition_validator/validator.cc

```diff
@@ -402,9 +404,108 @@ void validateFinalMethodHelper(const core::GlobalState &gs, const core::ClassOrM
             sym.name(gs) == core::Names::unresolvedAncestors()) {
             continue;
         }
-        if (auto e = gs.beginError(sym.loc(gs), core::errors::Resolver::FinalModuleNonFinalMethod)) {
+        auto defLoc = sym.loc(gs);
+        if (auto e = gs.beginError(defLoc, core::errors::Resolver::FinalModuleNonFinalMethod)) {
             e.setHeader("`{}` was declared as final but its method `{}` was not declared as final",
                         errMsgClass.show(gs), sym.name(gs).show(gs));
+
+            // We don't have a direct link to the signature for the method, so we
+            // heuristically look for the matching signature.  This heuristic
+            // assumes that sigs and method definitions appear on separate lines
+            // and would get tripped up by something like:
+            //
+            // sig {...}; def foo; ...; end
+            // def bar; ... end
```

Comment:
Would you mind adding a test for this case to document the incorrect behavior?

Reply:
I realized that this example is actually OK, because both methods aren't `final` in the first place, so the sig winds up with an autocorrect no matter what.  I'm leaning towards deleting the example.